frontend/js/pages/BattleDetail.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import CardTeam from 'pages/CardTeam';
+import { apiUrls, getFromApi, createTeamUrl } from 'utils/api';
+
+function BattleDetail() {
+  const [user, setCurrentUser] = useState();
+  const [battle, setBattle] = useState();
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    getCurrentUserData();
+    getTeamData();
+  }, []);
+
+  const getCurrentUserData = async () => {
+    const user = await getFromApi(apiUrls.currentUser);
+    setCurrentUser(user);
+    return user;
+  };
+  const getTeamData = async () => {
+    const data = await getFromApi(apiUrls.battleDetail(id));
+    setBattle(data);
+    return data;
+  };
+  let currentUserTeam = null;
+  let otherUserTeam = null;
+  if (!battle) {
+    return (
+      <div className="battle_container_detail">
+        <img
+          alt="battle not found"
+          className="img_detail"
+          src="http://www.i2softbd.com/template/TPL-007/images/404-Page-Not-Found.png"
+        />
+      </div>
+    );
+  }
+  if (battle.teams.length === 1) {
+    currentUserTeam = battle.teams[0].trainer.email === user.email ? battle.teams[0] : null;
+    otherUserTeam = currentUserTeam === null ? battle.teams[0] : null;
+  } else if (battle.teams.length === 2) {
+    currentUserTeam =
+      battle.teams[0].trainer.email === user.email ? battle.teams[0] : battle.teams[1];
+    otherUserTeam = currentUserTeam === battle.teams[0] ? battle.teams[1] : battle.teams[0];
+  }
```

Comment:
What do you think of extracting this to a function outside this file?

Reply:
It's a good idea, thanks!!